### DISCUSSION TITLE
F-Droid: Implement auto updates

### DISCUSSION THREAD

**Discussion by jakubgs** [status-im/status-legacy#16625] [General]
We have successfully deployed two versions of the Android app to F-Droid as part of #8512.

Right now the update process is manual and described in the [`doc/FDROID.md`](https://github.com/status-im/status-react/blob/develop/doc/FDROID.md) file. But it is possible to [automate this process](https://f-droid.org/en/docs/Update_Processing/) using the `AutoUpdateMode`, `UpdateCheckMode`, and `UpdateCheckIgnore` settings:

https://f-droid.org/en/docs/Build_Metadata_Reference/#AutoUpdateMode
https://f-droid.org/en/docs/Build_Metadata_Reference/#UpdateCheckMode
https://f-droid.org/en/docs/Build_Metadata_Reference/#UpdateCheckIgnore

The most sensible solution would be to use the `AutoUpdateMode: Version` and/or `UpdateCheckMode: Tags` setting.

~~The main problem to solve is how to match the `versionCode` with the one for the Play Store release. This is important in order to avoid F-Droid release being able to be installed on top of the Play Store one or vice versa depending on order.~~

As @lokalise-bot commented below, the `versionCode` issue might not be as serious.

**Comment by licaon-kter** [comment]
You can't install since Android doesn't allow this, different singnature.

**Comment by licaon-kter** [comment]
That's me the bot :)

Anyway, the `versionName` is more important as people tend to compare that between stores and then open issues with _"when is this going to be updated too"_

`versionCode` is usually not that visible.

**Comment by cammellos** [comment]
Converting to a discussion since it looks like it's not yet actionable and we need a bit more discussion to specify the work, feel free to reopen if necessary
